client/app/scripts/actions/app-actions.js

```diff
@@ -604,6 +622,31 @@ export function receiveNodesDelta(delta) {
   };
 }
 
+function updateFromNodesDeltaBuffer(dispatch, state) {
+  const isPaused = isPausedSelector(state);
+  const isBufferEmpty = state.get('nodesDeltaBuffer').isEmpty();
+
+  if (isPaused || isBufferEmpty) {
+    dispatch(resetNodesDeltaBuffer());
+  } else {
+    dispatch(receiveNodesDelta(state.get('nodesDeltaBuffer').first()));
+    dispatch({ type: ActionTypes.POP_NODES_DELTA_BUFFER });
+  }
+}
+
+export function clickResumeUpdate() {
+  return (dispatch, getServiceState) => {
+    dispatch({
+      type: ActionTypes.CLICK_RESUME_UPDATE
+    });
+    // Periodically merge buffered nodes deltas until the buffer is emptied.
+    nodesDeltaBufferUpdateTimer = setInterval(
```

Comment:
I wonder if 1000ms is slow enough to make the UI less sluggish during replay. Maybe even use requestAnimationFrame inside to trigger the next `updateFromNodesDeltaBuffer`